wgpu-hal/src/lib.rs

```diff
@@ -522,6 +535,60 @@ pub trait Adapter: WasmNotSendSync {
     unsafe fn get_presentation_timestamp(&self) -> wgt::PresentationTimestamp;
 }
 
+/// A connection to a GPU and a pool of resources to use with it.
+///
+/// A `wgpu-hal` `Device` represents an open connection to specific graphics
+/// processor, controlled via the backend [`Device::A`]. This corresponds to a
```

Comment:
style: It's nice to use the rename-proof `Self`, but definitely a matter of taste:

```suggestion
/// processor, controlled via the backend [`Self::A`]. This corresponds to a
```

Reply:
Interesting. I think it's clearer for readers to just say `Device`.